### PR TITLE
docs: test broken links in website

### PR DESCRIPTION
**Pull request by gastonfournier**
## About the changes
Adds broken link checks after building the website. 
Initially, the workflow is meant to be triggered manually

### CODE REVIEW

**Comment by thomasheartman** [comment]
@gastonfournier What's the status on this? 😄

File: .github/workflows/check_links.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          title: Link Checker Report
+          content-filepath: ${{ env.issue-content }}
+          # Update an existing issue if one was found (issue_number),
+          # otherwise an empty value creates a new issue:
+          issue-number: ${{ steps['last-issue']['outputs']['issue-number'] }}
+          # Add a label(s) that `last-issue` can use to find this issue,
+          # and any other relevant labels for the issue itself:
+          labels: |
+            ${{ env.issue-lookup-label }}
+            report
```
Comment (gastonfournier) [inline .github/workflows/check_links.yaml:62]:
```suggestion
            docs
```

File: .github/workflows/check_links.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        with:
+          path: .lycheecache
+          key: cache-lychee-${{ github.sha }}
+          restore-keys: cache-lychee-
+      - name: Link Checker
+        id: lychee
+        uses: lycheeverse/lychee-action@v1.5.0
+        with:
+          fail: true
+          args: '"./website/docs/**/*.md" "./src/mailtemplates/*.mustache" --scheme http --scheme https --cache --max-cache-age 1d --exclude-mail --verbose' # other excludes are in .lycheeignore
```
Comment (gastonfournier) [inline .github/workflows/check_links.yaml:35]:
```suggestion
          args: '"./website/docs/**/*.md" '"./website/docs/**/*.mdx" "./src/mailtemplates/*.mustache" --scheme http --scheme https --cache --max-cache-age 7d --exclude-mail --verbose' # other excludes are in .lycheeignore
```

File: .github/workflows/check_links.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+on:
+  workflow_dispatch:
+  push:
+  pull_request:
+    branches: [docs/find-broken-links]
+    paths:
+      - website/**
+      - .github/workflows/build_doc_prs.yaml
+      - .github/workflows/generate-docs.yaml
```
Comment (gastonfournier) [inline .github/workflows/check_links.yaml:11]:
```suggestion
```

File: .github/workflows/check_links.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        id: lychee
+        uses: lycheeverse/lychee-action@v1.5.0
+        with:
+          fail: true
+          args: '"./website/docs/**/*.md" "./website/docs/**/*.mdx" "./src/mailtemplates/*.mustache" --scheme http --scheme https --cache --max-cache-age 7d --exclude-mail --verbose' # other excludes are in .lycheeignore
+          output: ${{ env.issue-content }}
+
+      # Permissions (issues: read)
+      - name: 'Look for an existing issue'
+        if: ${{ false }}
```
Comment (gastonfournier) [inline .github/workflows/check_links.yaml:40]:
```suggestion
        if: ${{ failure() }}
```

File: .github/workflows/check_links.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        id: last-issue
+        uses: micalevisk/last-issue-action@v2
+        # Find the last updated open issue with a `automated-issue` label:
+        with:
+          state: open
+          labels: ${{ env.issue-lookup-label }}
+
+      # Permissions (issues: write)
+      - name: 'Create a new issue, or update an existing one'
+        if: ${{ false }}
```
Comment (gastonfournier) [inline .github/workflows/check_links.yaml:50]:
```suggestion
        if: ${{ failure() }}
```

**Review by gastonfournier — COMMENTED** [review]
Do create the issue

**Comment by gastonfournier** [comment]
Based on https://github.com/Unleash/unleash/runs/8228984635?check_suite_focus=true it does not make sense to run always

**Review by thomasheartman — APPROVED** [review]
Nice work! Let's get it in 🚂

**Pull request merged**